fix: add swagger whitelist and modify the swaggerServer port

zhangwt-cn opened the pull request:
fixed #18

josdem reviewed (changes requested):
Hi @zhangwt-cn thanks for sending a PR; I do not think this configuration is working as expected. I added a test case named `SwaggerTest` on branch `feature/18` with your changes; please check it out. Also, I added a Jenkins job to run test cases, and as you can see, it is sending a 401 status code. https://jenkins.josdem.io/job/shopping-cart-service/

zhangwt-cn commented:
> Hi @zhangwt-cn thanks for sending a PR; I do not think this configuration is working as expected. I added a test case named `SwaggerTest` on branch `feature/18` with your changes; please check it out. Also, I added a Jenkins job to run test cases, and as you can see, it is sending a 401 status code. https://jenkins.josdem.io/job/shopping-cart-service/

After reading the error report of the test case, I think this is a problem with your test case. In swagger v3, `/swagger-ui.html `will be redirected to `/swagger-ui/index.html`, so the test case should be modified as
```java
mockMvc.perform(get("/swagger-ui/index.html")).andExpect(status().isOk());
```
So the address to access swaager in the browser is `http://ip:port/swagger-ui/index.html`. If you still want to use `http://ip:port/swagger-ui.html` then you must add whitelist
```java 
private final String[] WHITE_PATH = {"/swagger-ui/**", "/v3/api-docs/**", "/swagger-ui.html"};
```

josdem commented:
> > Hi @zhangwt-cn thanks for sending a PR; I do not think this configuration is working as expected. I added a test case named `SwaggerTest` on branch `feature/18` with your changes; please check it out. Also, I added a Jenkins job to run test cases, and as you can see, it is sending a 401 status code. https://jenkins.josdem.io/job/shopping-cart-service/
> 
> After reading the error report of the test case, I think this is a problem with your test case. In swagger v3, `/swagger-ui.html `will be redirected to `/swagger-ui/index.html`, so the test case should be modified as
> 
> ```java
> mockMvc.perform(get("/swagger-ui/index.html")).andExpect(status().isOk());
> ```
> 
> So the address to access swaager in the browser is `http://ip:port/swagger-ui/index.html`. If you still want to use `http://ip:port/swagger-ui.html` then you must add whitelist
> 
> ```java
> private final String[] WHITE_PATH = {"/swagger-ui/**", "/v3/api-docs/**", "/swagger-ui.html"};
> ```

Hi @zhangwt-cn  as far as I know springdoc default Swagger URL is `/webjars/swagger-ui/index.html`  I modified the test case with the complete URL and still `401`. I will keep looking, at least we have a test case to validate the expected behavior.

zhangwt-cn commented:
> > > Hi @zhangwt-cn thanks for sending a PR; I do not think this configuration is working as expected. I added a test case named `SwaggerTest` on branch `feature/18` with your changes; please check it out. Also, I added a Jenkins job to run test cases, and as you can see, it is sending a 401 status code. https://jenkins.josdem.io/job/shopping-cart-service/
> > 
> > 
> > After reading the error report of the test case, I think this is a problem with your test case. In swagger v3, `/swagger-ui.html `will be redirected to `/swagger-ui/index.html`, so the test case should be modified as
> > ```
> > mockMvc.perform(get("/swagger-ui/index.html")).andExpect(status().isOk());
> > ```
> > 
> > 
> > So the address to access swaager in the browser is `http://ip:port/swagger-ui/index.html`. If you still want to use `http://ip:port/swagger-ui.html` then you must add whitelist
> > ```
> > private final String[] WHITE_PATH = {"/swagger-ui/**", "/v3/api-docs/**", "/swagger-ui.html"};
> > ```
> 
> Hi @zhangwt-cn as far as I know springdoc default Swagger URL is `/webjars/swagger-ui/index.html` I modified the test case with the complete URL and still `401`. I will keep looking, at least we have a test case to validate the expected behavior.

It seems that you did not take the advice I gave you seriously, so I decided to withdraw this PR.
